bigquery_etl/cli/backfill.py

```diff
@@ -0,0 +1,222 @@
+    ctx.ensure_object(dict)
+    ctx.obj["TMP_DIR"] = ctx.with_resource(tempfile.TemporaryDirectory())
+
+
+@backfill.command(
+    help="""Create a new backfill entry in the backfill.yaml file.  Create
+    a backfill.yaml file if it does not already exist.
+
+    Examples:
+
+    \b
+    ./bqetl backfill create moz-fx-data-shared-prod.telemetry_derived.deviations_v1 \\
+      --start_date=2021-03-01 \\
+      --end_date=2021-03-31 \\
+      --exclude=2021-03-03 \\
+    """,
+)
+@click.argument("qualified_table_name")
+@sql_dir_option
+@click.option(
+    "--start_date",
+    "--start-date",
+    "-s",
+    help="First date to be backfilled. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    required=True,
+)
+@click.option(
+    "--end_date",
+    "--end-date",
+    "-e",
+    help="Last date to be backfilled. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=datetime.today(),
+)
+@click.option(
+    "--exclude",
+    "-x",
+    multiple=True,
+    help="Dates excluded from backfill. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=None,
+)
+@click.option(
+    "--watcher",
+    "-w",
+    help="Watcher of the backfill (email address)",
+    default=DEFAULT_WATCHER,
+)
+@click.pass_context
+def create(
+    ctx,
+    qualified_table_name,
+    sql_dir,
+    start_date,
+    end_date,
+    exclude,
+    watcher,
+):
+    """CLI command for creating a new backfill entry in backfill.yaml file.
+
+    A backfill.yaml file will be created if it does not already exist.
+    """
+    try:
+        match = QUALIFIED_TABLE_NAME_RE.match(qualified_table_name)
+        project_id = match.group("project_id")
+        dataset_id = match.group("dataset_id")
+        table_id = match.group("table_id")
+    except AttributeError:
+        click.echo(
+            "Qualified table name must be named like:" + " <project>.<dataset>.<table>"
+        )
+        sys.exit(1)
+
+    path = Path(sql_dir)
+
+    query_path = path / project_id / dataset_id / table_id
+
+    if not query_path.exists():
+        click.echo(f"{project_id}.{dataset_id}.{table_id}" + " does not exist")
+        sys.exit(1)
+
+    backfill = Backfill(
+        entry_date=TODAY,
+        start_date=start_date.date(),
+        end_date=end_date.date(),
+        excluded_dates=[e.date() for e in list(exclude)],
+        reason=DEFAULT_REASON,
+        watchers=[watcher],
+        status=BackfillStatus.DRAFTING,
+    )
+
+    backfills = []
+
+    backfill_file = query_path / BACKFILL_FILE
+
+    if backfill_file.exists():
+        backfills = Backfill.entries_from_file(backfill_file)
+        for entry in backfills:
+            validate_duplicate_entry_dates(backfill, entry)
+            if entry.status == BackfillStatus.DRAFTING:
+                validate_overlap_dates(backfill, entry)
+
+    backfills.append(backfill)
```

Comment:
Should this be an `insert`? Since we want the newest backfill to be at the front?